speech.js

```diff
@@ -134,41 +192,194 @@ class Speech {
     this.apiKey = apiKey;
   }
 
-  async fetchVoices() {
-    return fetch(URL_VOICES, {
+  async fetchVoices(options = {}) {
+    const starred = options.starred || false;
+    const owner = options.owner || 'all';
+    let url = `${_BASE_URL}${_LIST_VOICES_ENDPOINT}?starred=${starred}&owner=${owner}`;
+    return fetch(url, {
+      headers: this._getHeaders(),
+      method: "GET",
+    })
+      .then(async (response) => {
+        await this._handle_response_errors(response);
+        return response.json();
+      })
+      .catch((error) => {
+        throw new Error(`[fetchVoices] ${error.message}`);
+      });
+  }
+
+  async fetchVoice(voice) {
+    const url = `${_BASE_URL}${_VOICE_ENDPOINT.replace('{id}', voice)}`;
+    return fetch(url, {
       headers: this._getHeaders(),
-      method: 'GET'
-    }).then(response => response.json());
+      method: "GET",
+    })
+      .then(async (response) => {
+        await this._handle_response_errors(response);
+        return response.json();
+      })
+      .catch((error) => {
+        throw new Error(`[fetchVoice] ${error.message}`);
+      });
   }
 
+  async createVoice(name, enhance, filenames, type = 'instant', gender = null, description = null) {
+    if (name.length === 0) {
+      throw new Error('[Speech.createVoice] Name must be non-empty.');
+    }
+    if (filenames.length === 0) {
+      throw new Error('[Speech.createVoice] Filenames must be non-empty.');
+    }
+
+    const metadata = JSON.stringify({
+      name,
+      enhance,
+      type,
+      gender,
+      description,
+    });
+
+    const formData = new FormData();
+    formData.append('metadata', metadata, {
+        headers: {
+          'Content-Type': 'application/json', 'Content-Disposition': 'form-data; name="metadata"'
+        }
+    });
+    filenames.forEach((filename) => {
+        formData.append('file_field', fs.createReadStream(filename), {
+            filename: path.basename(filename),
+            headers: {
+                'Content-Type': 'application/octet-stream', 'Content-Disposition': 'form-data; name="file_field"'
+            }
+        });
+    });
+
+    return axios.post(`${_BASE_URL}${_CREATE_VOICE_ENDPOINT}`, formData, {
+        headers: {
+            ...formData.getHeaders(),
+            ...this._getHeaders(),
+        },
+    })
+    .then((response) => {
+      if (response.status !== 200) {
+        throw new SpeechError(response.status, response.data);
+      }
+      return response.data;
+    })
+    .catch((error) => {
+      throw new Error(`[createVoice] ${error.message}`);
+    });
+  }
+
+  async updateVoice(voice, options = {}) {
+    const requestOptions = {
+      name: options.name || undefined,
```

Comment:
Feel like I never really see this (though I recognize you're looking to be thorough and explicit re: defaults), I think JS usually people would just leave the undefined as implicit, so, like, just:

```
name: options.name,
```
@zachoverflow is our resident JS expert. Similar for above/below where we specify `undefined` as param defaults/fallbacks.